nav2_rviz_plugins/src/goal_tool.cpp

```diff
@@ -45,6 +46,14 @@ void GoalTool::onInitialize()
 
 void
 GoalTool::onPoseSet(double x, double y, double theta)
+{
+  execution_thread_ = std::make_unique<std::thread>(&GoalTool::onPoseSetImpl, this,
```

Comment:
maybe no need to have `execution_thread_` as a member variable as it's detaching